_RFCs/0001-rfc-for-rfcs.md

```diff
@@ -0,0 +1,185 @@
+
+#################
+
+## Table of Contents
+[Table of Contents]: #table-of-contents
+
+  - [Intro](#core-plans-rfcs)
+  - [Table of Contents](#table-of-contents)
+  - [When you need to follow this process](#when-you-need-to-follow-this-process)
+  - [Before creating an RFC](#before-creating-an-rfc)
+  - [What the process is](#what-the-process-is)
+  - [The RFC life-cycle](#the-rfc-life-cycle)
+  - [Reviewing RFCs](#reviewing-rfcs)
+  - [Implementing an RFC](#implementing-rfcs)
+  - [RFC Postponement](#rfc-postponement)
+  - [Help this is all too informal!]
+  - [Attribution]
+
+# Core-Plans RFCs
+[Core-Plans RFCs]: #core-plans-rfcs
+
+A majority of contributor and maintainer work can be implemented and reviewed via [the normal GitHub pull request workflow](https://guides.github.com/introduction/flow/). These changes include bug fixes, documentation improvements, package version bumps, and adding a new core-plan. 
+
+Other changes by their nature are considered more complex, and we prefer to have an established process that these changes can be put through in order to provide the Habitat Maintainers and Community members those an opportunity to contribute to the design process and produce a consensus among the community.
+
+The "RFC" (request for comments) process is intended to provide a consistent path and expectation for a few things specifically: tooling changes, process changes, or new standards that might modify the acceptance criteria for new packages entering core-plans. The purpose is to ensure that all project stakeholders can be confident and aware of the direction core-plans and their maintenance are heading in.
+
+## When you need to follow this process
+[When you need to follow this process]: #when-you-need-to-follow-this-process
+
+You need to follow this process if you intend to make any change considered "substantial" to the Core-Plans repository, tooling, or maintenance processes (including the RFC process itself). What constitutes a "substantial" change may evolve over time based on the community, but may initially include the following.
+
+  - Any process or operational changes to the Core-Plans Maintenance Policies.
+  - Any package feature or rewrite that breaks backwards compatibility for current users.
+  - Any Removal of a package's features (see breaking change).
+  - Changes to the standardized operational tooling in use by Core-Plans maintainers.
+  - Any changes to the standard core-plans accepted style and baseline.
+  - Any inbound Habitat change that obligates all entries in Core-Plans to change.
+
+**NOTE:** If you have opened a single PR that changes multiple plans, it is assumed you are making a substantial change to core plans. Substantial change PRs submitted without an RFC will be closed and you will be directed to open an RFC. If this was not your intention (for example in the case of updating two interdependent packages) please open separate PRs per package change.
```

Comment:
I think I disagree. I am obviously 100% open to discussion on it but the impetus here is just that while I was thinking about the maintenance I've been doing on core-plans I found having separate PRs felt quite a bit more manageable than having a pr that lumps a bunch of things together.

The primary things that noticed were that separate PRs gives us clean and easily searchable history of merged changes that are package specific without obligating users to be intentional with their commit history. The other thing is with builder if you merge two interdependent changes and the jobs get scheduled in parallel on separate worker nodes you're going to end up with build failures that obligate you as a maintainer to do a bunch of digging to determine whether one of those failures was legitimate or just due to the parallelization. It also gives us a nice clean line for new maintainers to follow on how to handle incoming changes. Maybe this statement needs more refinement to make it less prescriptive for specific sub-cases? 

I don't want to make a change that is going to dissuade contribution though, I just want to make sure that maintainers don't end up in a situation where they could have had 4 package updated built and delivered but instead they have to do a bunch of detective work and debugging etc.

Reply:
> The other thing is with builder if you merge two interdependent changes and the jobs get scheduled in parallel on separate worker nodes you're going to end up with build failures that obligate you as a maintainer to do a bunch of digging to determine whether one of those failures was legitimate or just due to the parallelization. 

That is an interesting limitation of how bldr's github integration works.  It might be worth thinking about in the future.  However, for now I think it represents a good reason to generally prefer PR-per-plan in most cases.

Reply:
For the sake of further discussion, that outlined situation above is definitely a corner case. Like most of the time I don't think anyone would ever hit this but there is a chance. Now we also have the .bldr.toml file for noting explicit rebuild orders: for example in the case where we've got a package that sources another package but I don't know if the above situation would be caught by a bldr.toml entry

Reply:
Ok yes, I have validated that the dep graph is not traversed _before_ the jobs get scheduled. It's possible that something like that could get implemented in the future but for now the above statement of possible false positive failures holds true.